github/git_refs.go

```diff
@@ -61,8 +61,7 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 	r := new(Reference)
 	resp, err := s.client.Do(ctx, req, r)
 	if _, ok := err.(*json.UnmarshalTypeError); ok {
-		// Multiple refs, means there wasn't an exact match.
-		return nil, resp, nil
+		return nil, resp, fmt.Errorf("No exact match found for this ref")
```

Comment:
1. Keep the "// Multiple refs, means there wasn't an exact match." comment. It explains that getting `*json.UnmarshalTypeError` error means we're getting multiple refs. Without the comment, it's not clear.

2. Go style says error strings should begin with a lower case letter. See https://github.com/golang/go/wiki/CodeReviewComments#error-strings. So, it should be "no exact match found for this ref".

3. Since there's no dynamic formatting taking place, use `errors.New("no exact match found for this ref")` rater than `fmt.Errorf`.